module/zfs/dnode_sync.c

```diff
@@ -769,7 +764,8 @@ dnode_sync(dnode_t *dn, dmu_tx_t *tx)
 	 */
```

Comment:
I'd suggest referencing the new comment in `dnode_new_blkid()` regarding why the upper bit may be set here.